inbox/quick-response.xml

```diff
@@ -0,0 +1,212 @@
+
+<example><![CDATA[<message to="gitbot@example.com">
+  <body xml:lang="en">New merge request opened by ExampleUser: https://git.example.com/example/mrs/3/</body>
+  <available-action xmlns="]]>&ns;<![CDATA[" id="merge">
+    <label xml:lang="en">Merge Now</label>
+  </available-action>
+</message>]]></example>
+
+    <p>A single message MUST NOT contain multiple <tt>&lt;available-action&gt;</tt> elements with the same values for the <tt>@id</tt> attribute. Labels MUST be unique per id and language.</p>
+    <p>Clients that receive a message containing available actions SHOULD offer UI to select one of the actions.</p>
+  </section2>
+
+  <section2 anchor="selecting-an-action" topic="Selecting an Action">
+    <p>When the user selects an action, their client sends a message containing an <tt>&lt;action-selected&gt;</tt> element which identifies the selected action. The message does not contain a <tt>&lt;body&gt;</tt>.</p>
+
+<example><![CDATA[<message to="gitbot@example.com">
+  <action-selected xmlns="]]>&ns;<![CDATA[" id="merge" />
+</message>]]></example>
+  </section2>
+</section1>
+
+<section1 anchor="rules" topic="Business Rules">
+  <p>All message bodies SHOULD always list the (internationalized) allowed responses too, so that users of clients that don't support &xepname; can still know what the allowed responses are.</p>
+  <p>Bots or other entities that indicate allowed responses are free to accept variations of the response values for convenience, for example ignoring the casing or accepting abbreviations of response values.</p>
+  <p>Actions SHOULD only be a quicker way to access a feature that could also be accessed using information in the message body. For example, a bot that notifies about a new merge request includes in its notification message body a link to the web interface where manual merging is possible. An action could offer a more convenient way to merge, without taking the route via the web interface. In summary, users of clients that don't support &xepname; SHOULD still have a way to manually trigger the action.</p>
+  <p>Clients MUST only provide quick responses for the most recently received message that contains text content.</p>
+  <p>Clients MUST only provide action selection for the most recently received message that contains actions. This is slightly different than the rule for quick responses, in that the message defining the action does not have to be the most recent message with text content, but only the most recent message that contains actions.</p>
+  <p>Clients MAY decide to not offer action selection if the message containing the action is not the most recent message with text content.</p>
+  <p>It is up to the client and its specific UI requirements to decide what to do if both quick responses and actions are available, potentionally coming from different messages.</p>
```

Comment:
I feel like this should give some suggestions using RFC 2119 language since it's in the business rules section. It's not clear to my what I would do if I were a client dev just from reading this sentence.

Reply:
Would you say the same if I moved this to implementation notes? Not entirely sure what goes where.

Reply:
I'm really not sure what would be best, maybe someone else will have stronger opinions :)